src/cmd/shared.go

```diff
@@ -2,23 +2,26 @@ package cmd
 
 import (
 	"encoding/json"
+	"errors"
 	"fmt"
 	"log"
+	"strings"
 
 	"github.com/Originate/exosphere/src/aws"
 	"github.com/Originate/exosphere/src/types"
 	"github.com/Originate/exosphere/src/types/context"
 	"github.com/Originate/exosphere/src/types/deploy"
 )
 
-func getAwsConfig(appConfig types.AppConfig, profile string) types.AwsConfig {
+func getAwsConfig(appConfig types.AppConfig, remoteEnvironmentID string, profile string) types.AwsConfig {
+	appRemoteEnv := appConfig.Remote.Environments[remoteEnvironmentID]
 	return types.AwsConfig{
-		Region:               appConfig.Remote.Region,
-		AccountID:            appConfig.Remote.AccountID,
-		SslCertificateArn:    appConfig.Remote.SslCertificateArn,
+		Region:               appRemoteEnv.Region,
+		AccountID:            appRemoteEnv.AccountID,
+		SslCertificateArn:    appRemoteEnv.SslCertificateArn,
 		Profile:              profile,
-		SecretsBucket:        fmt.Sprintf("%s-%s-terraform-secrets", appConfig.Remote.AccountID, appConfig.Name),
-		TerraformStateBucket: fmt.Sprintf("%s-%s-terraform", appConfig.Remote.AccountID, appConfig.Name),
+		SecretsBucket:        fmt.Sprintf("%s-%s-%s-terraform-secrets", appRemoteEnv.AccountID, appConfig.Name, remoteEnvironmentID),
```

Comment:
I think it might be _slightly_ easier for the user to distinguish state buckets if they were prefixed with the remote environment ID? Since that's the only distinguishing factor between these names. Same for the `TerraformStateBucket`

Reply:
We could even consider creating a folder with the environment, i.e. `qa/id-appName-terraform-secrets`. That might provide even further separation, and I think that's what Terraform does when you use their built in environment manager

Reply:
If we need the account id in the bucket name, I think that should be first. This is going from highest to lowest level. 

An account id could deploy many applications and for each app we could deploy multiple environments.

If we do a folder structure, I think it should follow that as well. Why exactly did we need the account-id in the bucket name again?

Reply:
Because bucket names are global, so account id was an easy way to distinguish.

Ok, I think going with the hierarchical structure is good, but there doesn't seem to be a straightforward way to translate that to folder structure

Reply:
Global in what scope?

Reply:
Per discussion on standup going with the current setup